wireshark: 2.6.6 -> 3.0.1 [RDY]

teto opened the pull request:
###### Motivation for this change
full changelog 
https://code.wireshark.org/review/gitweb?p=wireshark.git;a=blob;f=ChangeLog;h=132c0093ac247b015dee713aee9bbdceafff9ac3;hb=937e33de60bcfcd6f68e7250e5e6914ae1d1e1e4

higher level changes: 
https://code.wireshark.org/review/gitweb?p=wireshark.git;a=blob;f=NEWS;h=e8da4fb667fde0ff6107c2c861f4cb7f33f3bfda;hb=937e33de60bcfcd6f68e7250e5e6914ae1d1e1e4

###### Things done
Removed gtk support


~~I end up with "cycle detected in the references of" mentioned in https://github.com/NixOS/nix/issues/1741 . I noticed the generated pkgconfig file $dev/wireshark.pc has a wrong libdir while its includedir is fine https://paste.ubuntu.com/p/tkM277HCdv/~~
Problem fixed, cmake files were referring to $dev. I moved them to $dev.


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

teto committed:
wireshark: 2.6.6 -> 3.0.1

See https://www.wireshark.org/docs/relnotes/wireshark-3.0.0.html.
Drops gtk build.

teto commented on pkgs/applications/networking/sniffers/wireshark/default.nix:
```diff
@@ -1,45 +1,42 @@
+, gnutls, libgcrypt, libgpgerror, geoip, openssl, lua5, python3, libcap, glib
 , libssh, zlib, cmake, extra-cmake-modules, fetchpatch, makeWrapper
-, withGtk ? false, gtk3 ? null, librsvg ? null, gsettings-desktop-schemas ? null, wrapGAppsHook ? null
+, withGtk ? false, librsvg ? null, gsettings-desktop-schemas ? null, wrapGAppsHook ? null
```
done.

teto commented:
An additionnal question is how to deal with the disappearing wireshark-gtk package ?

I wouldn't recommand to merge as-is as I haven't quite tested the binaries yet.

teto commented:
I've done a bit of testing and basic usage looks fine

lsix commented:
Hi,

There is the following line in `all-packages.nix` for `wireshark-qt`:

```nix
    qt5 = qt59;
```

Can you check if it is still necessary to pin a particular version of QT?

teto commented:
@lsix doesn't seem necessary anymore, I removed the mention.

teto commented:
callout to darwin users to break the cycle ref

teto commented:
As evoked on #nixos-dev, would be cool to have a hand from @NixOS/darwin-maintainers.

teto commented:
@LnL7 could you ping the darwin-maintainers team please ? It seems like I don't have the rights :(

LnL7 committed:
wireshark: fix darwin build

Fixing the cycle in the cmake files is also needed on darwin.

LnL7 committed:
wireshark: fix Wireshark.app on darwin

Generalize install_name_tool fixup calls for Wireshark.app to avoid breaking
when version numbers change.

LnL7 commented on pkgs/applications/networking/sniffers/wireshark/default.nix:
```diff
@@ -98,7 +87,12 @@ in stdenv.mkDerivation {
-  '';
+
+  '' + ''
+      # to remove "cycle detected in the references"
```
This is part of the else clause, so it didn't run on darwin.

teto replied:
arg my bad :/

LnL7 replied:
the precedence of if vs + is as little unexpected.

teto commented:
Just saw they released 3.0.1. https://www.wireshark.org/download.html so bumped the version number. Hopefully, it doesn't break build (built locally) and we can merge this.

teto commented:
thanks !

volth commented:
can we restore `wireshark` 2.6.x ?
It is still active upstream (`2.6.11` [released just 5 days ago](https://github.com/wireshark/wireshark/releases)) and `libwiretap` 2.x is a dependency of `panda` (not packaged yet in `nixpkgs`, I plan to do it in few days)

teto commented:
what is panda ?

volth commented:
> what is panda ?

https://github.com/panda-re/panda
